### ISSUE TITLE
Debug.Trace infinite loop

### ISSUE THREAD

**Issue by anonymous-human** [typelead/eta#757]
The following program:

```
module Main where
import Debug.Trace
main = putStrLn $ trace "Fred" "Test"
```

:will fail with this output:

```
Exception in thread "main" java.lang.StackOverflowError
	at eta.base.Utils.debugBelch(Utils.java:198)
	at eta.base.Utils.debugBelch(Utils.java:198)
	at eta.base.Utils.debugBelch(Utils.java:198)
        ...
```

The offending code in https://github.com/typelead/eta/blob/master/libraries/base/java-utils/Utils.java calls itself:

```
public static void debugBelch(String format, String string) {
        debugBelch(format, string);
}
```

I would submit a one line pull request but I cannot get figure out how to get my locally built *eta* to be used by *etlas*, so therefore cannot test that replacing *debugBelch* with *RuntimeLogging.debugBelch* will actually fix it (but it probably will).

**Comment by jneira** [comment]
@anonymous-human thanks for report and tried to fix it
The cleanInstall script [should build etlas](https://github.com/typelead/eta/blob/master/cleaninstall.sh#L17) and [make it work with the local version of eta](https://github.com/typelead/eta/blob/master/shake/Build.hs#L181)

**Comment by rahulmutt** [comment]
@anonymous-human Are you planning on submitting a PR for this?

**Comment by anonymous-human** [comment]
Hi Rahul,

I don't think I am, but not for lack of trying. When I go to build my copy of eta using *./cleaninstall.sh*, I get a compiler error on:

https://github.com/typelead/eta/blob/964a98bd6ead37fab1fd10d091b83aa3d2775065/libraries/base/java-utils/Utils.java#L164

```
        return new String(eta.ghc_prim.Utils.byteBufferToBytes(address, len), "UTF-8");
```

The compiler cannot find the package eta.ghc_prim. I am no Java expert, so that is where I got stuck unfortunately. I pulled from the *master* branch.

**Comment by rahulmutt** [comment]
@anonymous-human That error happens with Java 9 and above, although I recently pushed a fix. It may also happen with an unsupported locale.

Try this and let me know:
```
$ export LC_ALL=en_US.UTF-8
$ export LANG=en_US.UTF-8
```

Also if you pulled from `master` a while back, I suggest you update to get the Java 9+ fix.

```
$ git pull
$ git submodule update --init --recursive
```

**Comment by rahulmutt** [comment]
Fixed with #766
